### DISCUSSION TITLE
Compiling / pkgconfig / glibc

### DISCUSSION THREAD

**Discussion by cwmoriarty** [Entware/Entware#777] [Q&A] [answered]
I'm trying to do some compiling with Entware (x64) and pkg-config is blocking me during an autoconf job.

In /opt/lib I have all of the expected shared objects I installed with opkg (along with the other required build tools). 
eg:
libgio-2.0.so
libglib-2.0.so  

However pkgconfig only has a single pc file: `/opt/lib/pkgconfig/zlib.pc` 

I don't know much about pkg-config, but PKG_CHECK_MODULES is failing to find anything (is it looking for headers?)

Getting stuck in a configure job. 
https://github.com/cockpit-project/cockpit/blob/main/configure.ac#L85
```
PKG_CHECK_MODULES(glib, [gio-2.0 >= 2.56 gio-unix-2.0])
glib_CFLAGS="${glib_CFLAGS} -DGLIB_VERSION_MIN_REQUIRED=$GLIB_API_VERSION"
glib_CFLAGS="${glib_CFLAGS} -DGLIB_VERSION_MAX_ALLOWED=$GLIB_API_VERSION"

PKG_CHECK_MODULES(libsystemd, [libsystemd >= 235])
PKG_CHECK_MODULES(json_glib, [json-glib-1.0 >= 1.4])
PKG_CHECK_MODULES(gnutls, [gnutls >= 3.6.0])
PKG_CHECK_MODULES(krb5, [krb5-gssapi >= 1.11 krb5 >= 1.11])
if test "$enable_polkit" != "no"; then
```

Output:
```
...
checking for pkg-config... /opt/bin/pkg-config
checking pkg-config is at least version 0.9.0... yes
checking for gio-2.0 >= 2.56 gio-unix-2.0... no
configure: error: Package requirements (gio-2.0 >= 2.56 gio-unix-2.0) were not met:

No package 'gio-2.0' found
No package 'gio-unix-2.0' found

Consider adjusting the PKG_CONFIG_PATH environment variable if you
installed software in a non-standard prefix.

Alternatively, you may set the environment variables glib_CFLAGS
and glib_LIBS to avoid the need to call pkg-config.
See the pkg-config man page for more details.
```

**Comment by ryzhovau** [comment, marked as answer]
> In /opt/lib I have all of the expected shared objects I installed with opkg (along with the other required build tools).

All Entware packages build with cross-compilation environment. Native compilation provided as is with no any support and can be used for quite simple projects.
